basics section update for ecp21 tutorial

becker33 opened the pull request:
No major substance changes
Minor change to use buildcache API to get public key,
instead of `spack gpg trust /path/to/key`
Updates to use spack v0.16.1 or releases/v0.16

scheibelp commented on outputs/packaging/create.out:
```diff
@@ -1,6 +1,6 @@
 ==> This looks like a URL for mpileaks
 
-==> This package looks like it uses the autotools build system
+==> This package#looks like it uses the autotools build system
```
Does this command in fact output a `#` between "packages" and "looks"?

scheibelp commented on outputs/packaging/install-mpileaks-2.out:
```diff
@@ -1,4 +1,4 @@
-$ spack install mpileaks
+$ spack install --keep-stage mpileaks
```
Was this command change intentional?

scheibelp commented on outputs/modules/tcl-refresh-6.out:
```diff
@@ -1,2 +1,12 @@
 $ spack module tcl refresh -y
-==> Regenerating tcl module files
+==> Error: Name clashes detected in module files:
```
It looks like this command failed where it succeeded before.

scheibelp commented on outputs/modules/tcl-refresh-5.out:
```diff
@@ -1,2 +1,12 @@
 $ spack module tcl refresh --delete-tree -y
-==> Regenerating tcl module files
+==> Error: Name clashes detected in module files:
```
It looks like this command failed where it succeeded before.

scheibelp commented on outputs/modules/module-show-2.out:
```diff
@@ -1,19 +1,2 @@
-Fortran, Ada, and Go, as well as libraries for these languages.
-]])
-
+sh: 1: module: not found
```
It looks like this is failing where it was succeeding before

scheibelp commented on outputs/modules/module-show-1.out:
```diff
@@ -1,23 +1,2 @@
-Fortran, Ada, and Go, as well as libraries for these languages.
-]])
-
+sh: 1: module: not found
```
It looks like this is failing where it was succeeding before.

scheibelp commented on outputs/modules/module-avail-2.out:
```diff
@@ -1,36 +1,2 @@
-
-
+$ module avail gcc-8.3.0-gcc-7.5.0-6hbkzol
+sh: 1: module: not found
```
It looks like this is failing where it was succeeding before.

scheibelp commented on outputs/modules/module-avail-1.out:
```diff
@@ -1,21 +1,2 @@
-Use "module keyword key1 key2 ..." to search for all possible modules matching any of the "keys".
-
-
+sh: 1: module: not found
```
It looks like this is failing where it was succeeding before.

scheibelp commented on outputs/modules/load-direct.out:
```diff
@@ -1,5 +1,2 @@
-Autoloading py-numpy/1.19.0-gcc-8.3.0-openblas
-Autoloading python/3.7.7-gcc-8.3.0
-Autoloading python/3.7.7-gcc-8.3.0
+sh: 1: module: not found
```
It looks like this is failing where it was succeeding before.

scheibelp commented on outputs/environments/env-status-2.out:
```diff
@@ -1,4 +1,4 @@
-$ despacktivate      # short alias for 'spack env deactivate'
+$ despacktivate      short alias for
```
It looks like there's a typo here (or prematurely-truncated string)

scheibelp commented on outputs/environments/config-get-1.out:
```diff
@@ -1,10 +1,10 @@
-#
-# It describes a set of packages to be installed, along with
-# configuration settings.
+  This is a Spack Environment file.
```
I think these `#` characters were removed incorrectly

scheibelp commented on outputs/environments/anonymous-create-2.out:
```diff
@@ -1,11 +1,11 @@
-#
-# It describes a set of packages to be installed, along with
-# configuration settings.
+  This is a Spack Environment file.
```
I think these `#` characters were removed incorrectly

scheibelp commented on outputs/environments/add-anonymous-1.out:
```diff
@@ -2,12 +2,12 @@ $ spack add hdf5@5.5.1
-#
-# It describes a set of packages to be installed, along with
-# configuration settings.
+  This is a Spack Environment file.
```
I think these `#` characters were removed incorrectly

scheibelp commented on outputs/basics/uninstall-zlib.out:
```diff
@@ -1,8 +1,8 @@
 $ spack uninstall zlib %gcc@6.5.0
+y
```
where does this 'y' come from?

becker33 replied:
It comes from piping `y` into the command to generate the output in a script. I'll fix those.

scheibelp reviewed (changes requested):
Biggest issue is that some commands appear to be failing: e.g. the output of `outputs/modules/module-avail-2.out` is now an error message vs. output.

Other than that the remaining issues seem minor (typos or minor errors in copying the output).

becker33 commented:
@scheibelp I believe I've addressed all review comments.

Several of your comments were on files I hadn't intended to modify in this PR. Those files have been removed.
